test/index.spec.js

```diff
@@ -4,55 +4,71 @@ import undoable, { ActionCreators, excludeAction, includeAction, isHistory } fro
 
 const decrementActions = ['DECREMENT']
 
-const testConfigZero = {
-  FOR_TEST_ONLY_includeActions: decrementActions,
-  filter: includeAction(decrementActions)
-}
-
-const testConfigOne = {
-  limit: 100,
-  initTypes: 'RE-INITIALIZE',
-  FOR_TEST_ONLY_excludedActions: decrementActions,
-  filter: excludeAction(decrementActions)
-}
-const initialStateOne = {
-  past: [0, 1, 2, 3],
-  present: 4,
-  future: [5, 6, 7]
-}
-
-const testConfigTwo = {
-  limit: 1024,
-  initTypes: 'RE-INITIALIZE'
-}
-const initialStateTwo = {
-  past: [123],
-  present: 5,
-  future: [-1, -2, -3]
-}
-
-const testConfigThree = {
-  limit: -1,
-  initTypes: []
-}
-const initialStateThree = {
-  past: [5, {}, 3, null, 1],
-  present: Math.pow(2, 32),
-  future: []
-}
-
-runTestWithConfig(undefined, undefined, 'Default config')
-runTestWithConfig({ initTypes: [] }, undefined, 'No Init types')
-runTestWithConfig({ limit: 200 }, 100, 'Initial State equals 100')
-runTestWithConfig({}, {'present': 0}, 'Initial State that looks like a history')
-runTestWithConfig(testConfigZero, undefined, 'Filter (Include Actions)')
-runTestWithConfig(testConfigOne, initialStateOne, 'Initial History and Filter (Exclude Actions)')
-runTestWithConfig(testConfigTwo, initialStateTwo, 'Initial State and Init types')
-runTestWithConfig(testConfigThree, initialStateThree, 'Erroneous configuration')
+runTests('Default config')
+runTests('No Init types', {
+  unduableConfig: {
+    initTypes: []
+  }
+})
+runTests('Initial State equals 100', {
+  unduableConfig: {
+    limit: 200
+  },
+  initialStoreState: 100
+})
+runTests('Initial State that looks like a history', {
+  unduableConfig: {},
+  initialStoreState: {'present': 0}
+})
+runTests('Filter (Include Actions)', {
+  unduableConfig: {
+    filter: includeAction(decrementActions)
+  },
+  testConfig: {
+    includeActions: decrementActions
+  }
+})
+runTests('Initial History and Filter (Exclude Actions)', {
+  unduableConfig: {
+    limit: 100,
+    initTypes: 'RE-INITIALIZE',
+    filter: excludeAction(decrementActions)
+  },
+  initialStoreState: {
+    past: [0, 1, 2, 3],
+    present: 4,
+    future: [5, 6, 7]
+  },
+  testConfig: {
+    excludedActions: decrementActions
+  }
+})
+runTests('Initial State and Init types', {
+  unduableConfig: {
+    limit: 1024,
+    initTypes: 'RE-INITIALIZE'
+  },
+  initialStoreState: {
+    past: [123],
+    present: 5,
+    future: [-1, -2, -3]
+  }
+})
+runTests('Erroneous configuration', {
+  unduableConfig: {
+    limit: -1,
+    initTypes: []
+  },
+  initialStoreState: {
+    past: [5, {}, 3, null, 1],
+    present: Math.pow(2, 32),
+    future: []
+  }
+})
 
 // Test undoable reducers as a function of a configuration object
 // `label` describes the nature of the configuration object used to run a test
-function runTestWithConfig (testConfig, initialStoreState, label) {
+function runTests (label, { unduableConfig, initialStoreState, testConfig } = {}) {
```

Comment:
It's to add test specific test config, like which actions are excluded. I thought it made more sense than adding it to the config which is also used as undoable config.